### PR TITLE
Update Extensions

### PR DESCRIPTION
**Pull request by BobbyRBruce**
Added extensions:
- The Markdown linter.
- VSCode Liveshare

Removed extensions:
- Python Black - not needed or used.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -20,9 +20,10 @@
                 "ms-vscode.cpptools-themes",
                 "ms-vscode.makefile-tools",
                 "Tsinghua-Hexin-Joint-Institute.gem5-slicc",
-                "ms-python.black-formatter",
```
Comment (powerjg) [inline .devcontainer/devcontainer.json:23]: How is this not used? Shouldn't we be all using this to format our python code?

Does some other extension supersede this?

Comment (powerjg) [reply .devcontainer/devcontainer.json:23]: @BobbyRBruce I am cleaning out these PRs. Can you answer this question?

Comment (BobbyRBruce) [reply .devcontainer/devcontainer.json:23]: this is a vscode extension, not the the python black utility which is used in pre-commit. The pre-commit python black is installed via pre-commit.

Comment (BobbyRBruce) [reply .devcontainer/devcontainer.json:23]: This is not all that important. This is simply an unnecessary extension that isn't worth installing. To my knowledge it does no harm.

Comment (powerjg) [reply .devcontainer/devcontainer.json:23]: I guess I use it in vscode when I "format document" before committing. I like to see the formatted document before pre-commit makes its changes.

Comment (BobbyRBruce) [reply .devcontainer/devcontainer.json:23]: I never use it personally but if you do that's fine. I'll quickly update the PR.

Comment (BobbyRBruce) [reply .devcontainer/devcontainer.json:23]: Done.

**Review by powerjg — APPROVED** [review]
Thanks, Bobby!

**Pull request merged**